src/inet/RawEndPoint.cpp

```diff
@@ -424,7 +424,7 @@ INET_ERROR RawEndPoint::Listen(IPEndPointBasis::OnMessageReceivedFunct onMessage
 
 #if CHIP_SYSTEM_CONFIG_USE_SOCKETS
     // Wait for ability to read on this endpoint.
-    mSocket.SetCallback(HandlePendingIO, this);
+    mSocket.SetCallback(HandlePendingIO, reinterpret_cast<intptr_t>(this));
```

Comment:
This requires a more casts and stronger casts, I wonder if it is worth it?

Reply:
This was a request in the previous review, so you and Boris can fight it out :-)

Reply:
It was a suggestion; Boris said he was OK either way. Above was just a question. Ultimately I think it is up to you.

Reply:
Yeah, given what this diff looks like and that we have no non-pointer use cases for this closure, we should probably leave it as `void*` for now.  If we ever end up with non-pointer use cases, we can think about how to be best address those then...

Reply:
Added to my TODO list.